File: app/src/main/java/org/deiverbum/app/data/entity/BibleHomilyWithAll.java
package org.deiverbum.app.data.entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.model.ComentarioBiblico;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */
public class BibleHomilyWithAll {
    @Embedded
    public BibleHomilyJoinEntity bibliaLecturaEntity;

    @Relation(
            parentColumn = "homilyFK",
            entityColumn = "homilyFK",
            entity = BibleHomilyThemeEntity.class
    )

    public BibleHomilyThemeEntity themeEntity;

    @Relation(
            parentColumn = "homilyFK",
            entityColumn = "homilyID",
            entity = HomilyEntity.class
    )

    public HomilyAll homilia;
/*
    @Relation(
            parentColumn = "obraFK",
            entityColumn = "obraId",
            entity = PaterOpusEntity.class
    )
    public PaterOpusAll paterOpusAll;
*/


    public ComentarioBiblico getDomainModel() {
        ComentarioBiblico theModel=new ComentarioBiblico();
        if(themeEntity!=null) {
            theModel.setCita(themeEntity.getBiblical());
            theModel.setTema(themeEntity.getTheological());
            theModel.setRef(themeEntity.getReference());
        }
        theModel.setPadre(homilia.paterOpusAll.getPadre());
        theModel.setObra(homilia.paterOpusAll.obra.getObra());
        theModel.setTexto(homilia.homilia.getTexto());
        theModel.setFecha(String.valueOf(homilia.homilia.fecha));

        return theModel;
    }


}

File: app/src/main/java/org/deiverbum/app/core/database/model/relation/BibleHomilyWithAll.kt
package org.deiverbum.app.core.database.model.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.core.database.model.entity.BibleHomilyJoinEntity
import org.deiverbum.app.core.database.model.entity.BibleHomilyThemeEntity
import org.deiverbum.app.core.database.model.entity.HomilyEntity
import org.deiverbum.app.core.model.data.traditio.BibleComment
import org.deiverbum.app.core.model.data.traditio.Homily

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
data class BibleHomilyWithAll(
    @Embedded
    val bibliaLecturaEntity: BibleHomilyJoinEntity,

    @Relation(
        parentColumn = "homilyFK",
        entityColumn = "homilyFK"
    )
    val themeEntity: BibleHomilyThemeEntity?,

    @Relation(parentColumn = "homilyFK", entityColumn = "homilyID", entity = HomilyEntity::class)
    val homilia: HomilyAll
)

fun BibleHomilyWithAll.asExternalModel() = BibleComment(
    homilia.paterOpusAll.asExternalModel(),
    themeEntity?.reference!!,
    themeEntity.theological!!,
    themeEntity.biblical!!,
    homilia.homilia.fecha,
    homilia.homilia.texto
)

fun BibleHomilyWithAll.asExternalModelNew() = Homily(
    homilia.homilia.texto,
    homilia.homilia.fecha,
    homilia.paterOpusAll.asExternalModel(),
    //themeEntity.reference!!,
    //themeEntity.theological!!,
    themeEntity!!.biblical!!,
)


